$MirOS: ports/databases/db/patches/patch-dist_configure_ac,v 1.1 2005/05/19 19:58:29 bsiegert Exp $
--- dist/configure.ac.orig	Wed Oct 15 02:55:17 2003
+++ dist/configure.ac	Tue Jul 26 21:19:23 2005
@@ -313,11 +313,13 @@ JMODSUFFIX_CONFIG
 
 INSTALLER="\$(LIBTOOL) --mode=install cp -p"
 
+LT_CURRENT=`expr ${DB_VERSION_MAJOR} + ${DB_VERSION_MINOR}`
+
 MAKEFILE_CC="\$(LIBTOOL) --mode=compile ${MAKEFILE_CC}"
-MAKEFILE_SOLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CCLINK} -avoid-version"
+MAKEFILE_SOLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CCLINK} -version-info ${LT_CURRENT}:0:${DB_VERSION_MINOR}"
 MAKEFILE_CCLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CCLINK}"
 MAKEFILE_CXX="\$(LIBTOOL) --mode=compile ${MAKEFILE_CXX}"
-MAKEFILE_XSOLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CXXLINK} -avoid-version"
+MAKEFILE_XSOLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CXXLINK} -version-info ${LT_CURRENT}:0:${DB_VERSION_MINOR}"
 MAKEFILE_CXXLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CXXLINK}"
 
 LIBTOOL="\$(SHELL) ./libtool"
